Assets/Scripts/Inventory/InventoryDisplay.cs

```diff
@@ -70,4 +78,136 @@ public void displayDetails(int id)
         descriptionText.text = items[id].description;
         iconLarge.sprite = items[id].icon;
     }
+
+    public void CloseContextMenu()
+    {
+        if (menu.active)
+            menu.Hide();
+    }
+
+    public void displayContext(int id)
+    {
+        Logger.log("Open context for id: " + id);
+        if (menu.active)
+            menu.Hide();
+        menu.feed(items[id]);
+        menu.Show();
+    }
+
+    public void refreshDisplay()
+    {
+        items = inventory.getItems();
+        Logger.log("Items in inventory: " + items.Length);
+
+        items = SortInventory(items);
+
+        for (int i = 0; i < items.Length; i++)
+        {
+            GameObject instance = Instantiate(slotPrefab, slotParent);
+            instance.GetComponent<Image>().sprite = items[i].icon;
+            int ii = i;
+            //instance.GetComponent<Button>().onClick.AddListener(() => displayDetails(ii));
+            ButtonClick click = instance.GetComponent<ButtonClick>();
+            click.leftClick.AddListener(() => displayDetails(ii));
+            click.rightClick.AddListener(() => displayContext(ii));
+            instance.GetComponent<InventorySlot>().Display(items[i]);
+            Logger.log("Added listener with ID: " + i);
+        }
+
+        slotText.text = "Items: " + inventory.GetSlotsUsed() + "/" + playerProperties.slotCapacity;
+        weightText.text = "Capacity: " + playerProperties.weight + "/" + playerProperties.weightCapacity;
+
+        float gradientTime = playerProperties.GetSpeedPenaltyGradient() / 100;
+        Logger.log("Time: " + gradientTime);
+        weightText.color = weightTextGradient.Evaluate(gradientTime);
+
+        slotText.gameObject.SetActive(playerProperties.GetSlotCapacityEnabled());
+        weightText.gameObject.SetActive(playerProperties.GetWeightCapacityEnabled()); //Only show texts if type capacity is enabled
+    }
+
+    public InventoryItem[] SortInventory(InventoryItem[] items)
+    {
+        return SortInventory(items, sortMethod, sortDescending);
+    }
+
+    private InventoryItem[] SortInventory(InventoryItem[] items, InventorySortMethod newMethod, bool invert = false)
```

Comment:
Why not use a existing function, Array.Sort?
https://stackoverflow.com/questions/1304278/how-to-sort-an-array-containing-class-objects-by-a-property-value-of-a-class-ins